UBUNTU: xen: Do not use vfb device in HVM definitions

When talking to xend to create or modify an HVM instance,
graphics should be defined outside (without) any vfb device.
(see http://lists.xen.org/archives/html/xen-users/2011-11/msg00119.html)

The vfb definition works just out of luck, but doing so also defines
a vkbd device which causes xenbus init to fail after a long
time retrying.

Signed-off-by: Stefan Bader <stefan.bader@canonical.com>
Index: libvirt-0.9.8/src/xenxs/xen_xm.c
===================================================================
--- libvirt-0.9.8.orig/src/xenxs/xen_xm.c	2012-04-03 17:09:59.000000000 +0200
+++ libvirt-0.9.8/src/xenxs/xen_xm.c	2012-04-04 14:36:38.434001687 +0200
@@ -1669,7 +1669,7 @@ virConfPtr xenFormatXM(virConnectPtr con
     }
 
     if (def->ngraphics == 1) {
-        if (xendConfigVersion < (hvm ? 4 : XEND_CONFIG_MIN_VERS_PVFB_NEWCONF)) {
+        if (hvm || (xendConfigVersion < XEND_CONFIG_MIN_VERS_PVFB_NEWCONF)) {
             if (def->graphics[0]->type == VIR_DOMAIN_GRAPHICS_TYPE_SDL) {
                 if (xenXMConfigSetInt(conf, "sdl", 1) < 0)
                     goto no_memory;
Index: libvirt-0.9.8/src/xenxs/xen_sxpr.c
===================================================================
--- libvirt-0.9.8.orig/src/xenxs/xen_sxpr.c	2012-04-03 17:09:59.000000000 +0200
+++ libvirt-0.9.8/src/xenxs/xen_sxpr.c	2012-04-04 17:41:41.228280488 +0200
@@ -2296,9 +2296,8 @@ xenFormatSxpr(virConnectPtr conn,
             }
         }
 
-        /* PV graphics for xen <= 3.0.4, or HVM graphics for xen <= 3.1.0 */
-        if ((!hvm && xendConfigVersion < XEND_CONFIG_MIN_VERS_PVFB_NEWCONF) ||
-            (hvm && xendConfigVersion < 4)) {
+        /* PV graphics for xen <= 3.0.4, or HVM graphics */
+        if (hvm || (xendConfigVersion < XEND_CONFIG_MIN_VERS_PVFB_NEWCONF)) {
             if ((def->ngraphics == 1) &&
                 xenFormatSxprGraphicsOld(def->graphics[0],
                                          &buf, xendConfigVersion) < 0)
@@ -2327,10 +2326,8 @@ xenFormatSxpr(virConnectPtr conn,
     if (xenFormatSxprAllPCI(def, &buf) < 0)
         goto error;
 
-    /* New style PV graphics config xen >= 3.0.4,
-     * or HVM graphics config xen >= 3.0.5 */
-    if ((xendConfigVersion >= XEND_CONFIG_MIN_VERS_PVFB_NEWCONF && !hvm) ||
-        (xendConfigVersion >= 4 && hvm)) {
+    /* New style PV graphics config xen >= 3.0.4 */
+    if (!hvm && (xendConfigVersion >= XEND_CONFIG_MIN_VERS_PVFB_NEWCONF)) {
         if ((def->ngraphics == 1) &&
             xenFormatSxprGraphicsNew(def->graphics[0], &buf) < 0)
             goto error;
